deeplearning4j-core/src/main/java/org/deeplearning4j/util/StringGrid.java

```diff
@@ -404,7 +404,7 @@ private void modifyRows(Set<Integer> alreadyDeDupped,Integer column,List<Integer
     private String maximalValue(Map<String,Integer> map) {
         Counter<String> counter = new Counter<>();
         for (Map.Entry<String, Integer> entry : map.entrySet()) {
-            counter.incrementCount(entry.getKey(), map.get(entry.getValue()));
+            counter.incrementCount(entry.getKey(), entry.getValue());
```

Comment:
Clearly intended to access the integer value, which can't also be a key in the map (which are strings)